cargo install from source fails

Hey,

using the pre-built binary worked perfectly but running `cargo install --git https://github.com/prefix-dev/pixi.git` caused the following output

```bash
   Compiling pixi v0.0.2 (/Users/ytatar/.cargo/git/checkouts/pixi-2602b418e59cbe0c/3e10b5a)
error[E0308]: mismatched types
   --> src/repodata.rs:129:9
    |
127 |     let result = fetch::fetch_repo_data(
    |                  ---------------------- arguments to this function are incorrect
128 |         channel.platform_url(platform),
129 |         client,
    |         ^^^^^^ expected struct `rattler_networking::AuthenticatedClient`, found struct `reqwest::Client`
    |
note: function defined here
   --> /Users/ytatar/.cargo/git/checkouts/rattler-3fdeff0ba0214908/a697958/crates/rattler_repodata_gateway/src/fetch/mod.rs:249:14
    |
249 | pub async fn fetch_repo_data(
    |              ^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> src/environment.rs:538:21
    |
535 |                 .get_or_fetch_from_url(
    |                  --------------------- arguments to this function are incorrect
...
538 |                     download_client.clone(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^ expected struct `rattler_networking::AuthenticatedClient`, found struct `reqwest::Client`
    |
note: associated function defined here
   --> /Users/ytatar/.cargo/git/checkouts/rattler-3fdeff0ba0214908/a697958/crates/rattler/src/package_cache.rs:179:18
    |
179 |     pub async fn get_or_fetch_from_url(
    |                  ^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `pixi` due to 2 previous errors
error: failed to compile `pixi v0.0.2 (https://github.com/prefix-dev/pixi.git#3e10b5a5)`, intermediate artifacts can be found at `/var/folders/jb/1wt_13hj6hnfb8zry_121wcc0000gn/T/cargo-installAiuAH0`
```

Am I maybe misunderstanding something regarding the installation instructions?

--- a/docs/README.md
+++ b/docs/README.md
@@ -87,7 +87,7 @@ tested with cargo.
 To start using `pixi` from a source build run:
 
 ```shell
-cargo install --git https://github.com/prefix-dev/pixi.git
+cargo install --locked --git https://github.com/prefix-dev/pixi.git
 ```
 
 or when you want to make changes use:

